xbmc/addons/AddonManager.cpp

```diff
@@ -1098,30 +1029,12 @@ AddonOriginType CAddonMgr::GetAddonOriginType(const AddonPtr& addon) const
     return AddonOriginType::MANUAL;
 }
 
-void CAddonMgr::AddAddonIfLatest(const AddonPtr& addonToAdd,
-                                 std::map<std::string, AddonPtr>& map) const
+bool CAddonMgr::IsAddonDisabledWithReason(const std::string& ID,
+                                          AddonDisabledReason disabledReason) const
 {
-  const auto& latestKnown = map.find(addonToAdd->ID());
-  if (latestKnown == map.end() || addonToAdd->Version() > latestKnown->second->Version())
-    map[addonToAdd->ID()] = addonToAdd;
-}
-
-bool CAddonMgr::FindAddonAndCheckForUpdate(const AddonPtr& addonToCheck,
-                                           const std::map<std::string, AddonPtr>& map,
-                                           VECADDONS& vecAddons) const
-{
-  const auto& remote = map.find(addonToCheck->ID());
-  if (remote != map.end()) // is addon in the desired map?
-  {
-    if (remote->second->Version() > addonToCheck->Version())
-    {
-      // queue addon update
-      vecAddons.emplace_back(remote->second);
-    }
-    return true;
-  }
-
-  return false;
+  CSingleLock lock(m_critSection);
+  const auto disabledAddon = m_disabled.find(ID);
```

Comment:
`const auto&` maybe?

Reply:
can we have a reference to a function call? i'll try it

Reply:
works for sure